tools/apicoverage/apicoverage.go

```diff
@@ -187,58 +186,69 @@ func getRelevantLogs(fields []string) *string {
 	return nil
 }
 
-func createCases(tcName string, covered map[string]int, notCovered map[string]int) []testgrid.TestCase {
-	var tc []testgrid.TestCase
-
+func addTestCases(tcName string, covered map[string]int, notCovered map[string]int, testSuite *junit.TestSuite) {
 	if (len(covered) == 0 && len(notCovered) == 0) {
-		return tc
+		return
 	}
 
 	var percentCovered = float32(100 * len(covered) / (len(covered) + len(notCovered)))
-	tp := []testgrid.TestProperty{testgrid.TestProperty{Name: apiCoverage, Value: percentCovered}}
-	tc = append(tc, testgrid.TestCase{Name: tcName, Properties: testgrid.TestProperties{Property: tp}, Fail: false})
+	testCase := junit.TestCase{Name: tcName}
+	testCase.AddProperty(apiCoverage, fmt.Sprintf("%f", percentCovered))
+	testSuite.TestCases = append(testSuite.TestCases, testCase)
 
 	for key, value := range covered {
-		tp := []testgrid.TestProperty{testgrid.TestProperty{Name: apiCoverage, Value: float32(value)}}
-		tc = append(tc, testgrid.TestCase{Name: tcName + "/" + key, Properties: testgrid.TestProperties{Property: tp}, Fail: false})
+		coveredTestCase := junit.TestCase{Name: tcName + "/" + key}
+		coveredTestCase.AddProperty(apiCoverage, fmt.Sprintf("%d", value))
+		testSuite.TestCases = append(testSuite.TestCases, coveredTestCase)
 	}
 
 	for key, value := range notCovered {
-		tp := []testgrid.TestProperty{testgrid.TestProperty{Name: apiCoverage, Value: float32(value)}}
-		tc = append(tc, testgrid.TestCase{Name: tcName + "/" + key, Properties: testgrid.TestProperties{Property: tp}, Fail: true})
+		notCoveredTestCase := junit.TestCase{Name: tcName + "/" + key}
+		notCoveredTestCase.AddProperty(apiCoverage, fmt.Sprintf("%d", value))
+		failureMessage := "failure"
+		notCoveredTestCase.Failure = &failureMessage
```

Comment:
Why not combining this line and the previous?

Reply:
this doesn't seem possible to me, thought about these 2 approaches:
- assign value directly by reference of `notCoveredTestCase.Failure`: it complains as this is a null pointer right now.
- get a pointer to `"failure"`: based on what I have read so far, it doesn't seem possible to get a pointer directly from a string literal.